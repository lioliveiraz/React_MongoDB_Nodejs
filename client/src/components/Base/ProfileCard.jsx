import React, { useState, useEffect } from "react";

function ProfileCard({ profile: { loading, profile, errors } }) {
  const [userProfile, setUserProfile] = useState(null);
  const [userInfo, setUserInfo] = useState({});
  const [socialMedia, setSocialMedia] = useState({});
  const [error, setError] = useState();

  useEffect(() => {
    async function getDeveloper() {
      setError(errors);
      setUserProfile(profile);

      if (!loading) {
        setUserInfo(profile.user);
        setSocialMedia(profile.social);
      }
    }
    getDeveloper();
  }, [profile]);

  return (
    <>
      {userProfile ? (
        <div>
          <img src={userInfo.avatar} alt={userInfo.name} />
          <h2>{userInfo.name}</h2>
          <p>{userProfile.bio}</p>
          <p>{userProfile.role}</p>
          {userProfile.skills.map((skill) => {
            <p>{skill}</p>;
          })}

          <a href={socialMedia.youtube}> Youtube</a>
          <a href={socialMedia.linkedin}> Linkedin</a>
          <a href={socialMedia.github}> Github</a>
        </div>
      ) : (
        <div>{error ? error : "loading"}</div>
      )}
    </>
  );
}

export default ProfileCard;
