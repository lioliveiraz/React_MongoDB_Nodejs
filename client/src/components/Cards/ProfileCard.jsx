import React, { useState, useEffect } from "react";

function ProfileCard({ profile: { loading, profile, status } }) {
  const [userProfile, setUserProfile] = useState(null);
  const [userInfo, setUserInfo] = useState({});
  const [socialMedia, setSocialMedia] = useState(null);

  useEffect(() => {
    async function getDeveloper() {
      setUserProfile(profile);

      if (!loading) {
        setUserInfo(profile.user);
        setSocialMedia(profile.social ? profile.social : null);
      }
    }
    getDeveloper();
  }, [profile]);

  if (!profile) {
    return <p>{status}</p>;
  }

  return (
    <>
      {userProfile && (
        <div>
          <img src={userInfo.avatar} alt={userInfo.name} />
          <h2>Name:{userInfo.name}</h2>
          <p>Bio:{userProfile.bio}</p>
          <p>role:{userProfile.role}</p>
          <ol>
            Skills:{" "}
            {userProfile.skills.map((skill, index) => (
              <li key={index}>{skill}</li>
            ))}
          </ol>
          <h3>Social</h3>
          {socialMedia ? (
            <div style={{ background: "turquoise", color: "white" }}>
              <a href={socialMedia.youtube}> Youtube</a>
              <a href={socialMedia.linkedin}> Linkedin</a>
              <a href={socialMedia.githubusername}> Github</a>
            </div>
          ) : null}
        </div>
      )}
    </>
  );
}

export default ProfileCard;
