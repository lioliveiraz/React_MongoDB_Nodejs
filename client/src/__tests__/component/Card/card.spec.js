import React from "react";
import { shallow } from "enzyme";
import ProfileCard from "./../../../components/Cards/ProfileCard";

describe("Profile Card", () => {
  let wrapper;
  const data = {
    profile: {
      loading: false,
      profile: {
        name: "John",
        bio: "lorem",
        role: "role",
        avatar: "avatar",
        skills: ["skill"],
        social: {
          youtube: "youtube",
          linkedin: "linkedin",
          github: "github",
        },
      },
    },
  };

  it("should match snapshot", async () => {
    wrapper = await shallow(<ProfileCard {...data} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
