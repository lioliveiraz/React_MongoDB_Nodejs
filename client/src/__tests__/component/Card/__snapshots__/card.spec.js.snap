// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Card should match snapshot 1`] = `ShallowWrapper {}`;

exports[`Profile Card should match snapshot 2`] = `""`;
